jupyterlab/labapp.py

```diff
@@ -227,19 +227,23 @@ def _validate(self, data, base_url, page_url, workspaces_url):
         if 'data' not in workspace:
             raise Exception('The `data` field is missing.')
 
-        # See if workspace name is given, inject into workspace.
+        # If workspace_name is set in config, inject the
+        # name into the workspace metadata.
         if self.workspace_name is not None:
-            workspace_id = ujoin(base_url, workspaces_url, self.workspace_name)
+            if self.workspace_name == "":
+                workspace_id = ujoin(base_url, 'lab')
```

Comment:
@ian-r-rose 

Okay, here is the updated logic to catch `--name=""`. The `workspace_id` becomes the default `/lab` endpoint.

Reply:
I think this should be `workspace_id = ujoin(base_url, page_url)`

cf. line 244 below

Reply:
👍 Fixed!